src/main/java/org/apache/mesos/hdfs/Scheduler.java

```diff
@@ -73,18 +81,178 @@ public void registered(SchedulerDriver driver, FrameworkID frameworkId, MasterIn
+          }
+          break;
+        // TODO (elingg) add a configurable number of data nodes
+        case DATA_NODES :
+          break;
+      }
+    } else {
+      log.warn(String.format("Don't know how to handle state=%s for taskId=%s",
+          status.getState(), status.getTaskId().getValue()));
+    }
+  }
+
+  @Override
+  public void resourceOffers(SchedulerDriver driver, List<Offer> offers) {
+    log.info(String.format("Received %d offers", offers.size()));
+
+    if (liveState.getStagingTasksSize() != 0) {
+      log.info("Declining offers because tasks are currently staging");
+      for (Offer offer : offers) {
+        driver.declineOffer(offer.getId());
+      }
+    } else {
+      boolean acceptedOffer = false;
+      for (Offer offer : offers) {
+        if (acceptedOffer) {
+          driver.declineOffer(offer.getId());
+        } else {
+          switch (liveState.getCurrentAcquisitionPhase()) {
+            case RECONCILING_TASKS :
+              if (reconciliationComplete()) {
+                correctCurrentPhase();
+                driver.declineOffer(offer.getId());
+              } else {
+                log.info("Declining offers while reconciling tasks");
+                driver.declineOffer(offer.getId());
+              }
+              break;
+            case JOURNAL_NODES :
+              if (tryToLaunchJournalNode(driver, offer)) {
+                acceptedOffer = true;
+              } else {
+                driver.declineOffer(offer.getId());
+              }
+              break;
+            case START_NAME_NODES :
+              if (tryToLaunchNameNode(driver, offer)) {
+                acceptedOffer = true;
+              } else {
+                driver.declineOffer(offer.getId());
+              }
+              break;
+            case FORMAT_NAME_NODES :
+              driver.declineOffer(offer.getId());
+              break;
+            case DATA_NODES :
+              if (tryToLaunchDataNode(driver, offer)) {
+                acceptedOffer = true;
+              } else {
+                driver.declineOffer(offer.getId());
+              }
+              break;
+          }
+        }
+      }
+    }
+  }
+
+  @Override
+  public void slaveLost(SchedulerDriver driver, SlaveID slaveId) {
+    log.info("Slave lost slaveId=" + slaveId.getValue());
+  }
+
+  @Override
+  public void run() {
+    FrameworkInfo.Builder frameworkInfo = FrameworkInfo.newBuilder()
+        .setName(conf.getFrameworkName())
+        .setFailoverTimeout(conf.getFailoverTimeout())
+        .setUser(conf.getHdfsUser())
+        .setRole(conf.getHdfsRole())
+        .setCheckpoint(true);
+
+    try {
+      FrameworkID frameworkID = persistentState.getFrameworkID();
+      if (frameworkID != null) {
+        frameworkInfo.setId(frameworkID);
+      }
+    } catch (InterruptedException | ExecutionException | InvalidProtocolBufferException e) {
+      log.error("Error setting framework id", e);
```

Comment:
s/setting/recovering/ since it's more likely a problem getting it from persistentState than setting it in the FrameworkInfo.